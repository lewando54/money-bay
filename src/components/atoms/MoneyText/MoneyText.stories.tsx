/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { Meta,StoryObj } from '@storybook/react-native'

import MoneyText from './MoneyText'

const meta = {
    component: MoneyText,
    argTypes: {
        // To write
    }
} satisfies Meta<typeof MoneyText>

export default meta
type Story = StoryObj<typeof MoneyText>

export const Default: Story = {
    args: {
        // To write
    }
}
