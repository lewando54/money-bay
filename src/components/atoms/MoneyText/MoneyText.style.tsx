import { StyleSheet } from 'react-native'

import { MAIN_DARK } from '@styling/GlobalTheme.style'
export default StyleSheet.create({
    moneyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end',
    },
    floatCurrencyContainer: {
        flexDirection: 'row',
        alignItems: 'flex-end'
    },
    minusSign: {
        fontSize: 20,
        color: MAIN_DARK
    }
})
