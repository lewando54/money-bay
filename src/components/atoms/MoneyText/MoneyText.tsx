import React from 'react'
import { StyleProp, View, ViewStyle } from 'react-native'

import { CustomText } from '@atoms'
import { TCurrency } from '@utils/currency'

import moneyTextStyle from './MoneyText.style'

import globalThemeStyle, { globalColors } from '@styling/GlobalTheme.style'

interface IMoneyTextProps {
    amount?: number
    currency?: TCurrency
    hasNumberFormatting?: boolean
    isCurrencyBold?: boolean
    fontSize?: number
    containerStyle?: StyleProp<ViewStyle>
}

export default function MoneyText({
    amount=10,
    currency={iso: 'usd', symbol: '$', left: true},
    hasNumberFormatting=false,
    isCurrencyBold=true,
    fontSize=20,
    containerStyle
}: IMoneyTextProps){
    const minusSign = amount < 0 && <CustomText style={[globalThemeStyle.text_Regular, moneyTextStyle.minusSign]}>- </CustomText> 
    const amountIntFinal = hasNumberFormatting ? Math.abs(Math.floor(amount)).toLocaleString('en-US').replace(',', ' ') : Math.abs(Math.floor(amount))
    const amountDecFinal = Math.abs(((amount * 100) % 100)).toLocaleString('en-US', {minimumIntegerDigits: 2})

    return (
        <View style={[moneyTextStyle.moneyContainer, containerStyle]}>
            {minusSign}
            <CustomText style={[globalThemeStyle.text_Regular, {fontSize: fontSize, color: globalColors.MAIN_DARK}]}>{amountIntFinal}.</CustomText>
            <View style={[moneyTextStyle.floatCurrencyContainer, {paddingBottom: fontSize / 10}]}>
                <CustomText style={[globalThemeStyle.text_Regular, {fontSize: Math.ceil(fontSize / 1.5), color: globalColors.MAIN_DARK}]}>{amountDecFinal}</CustomText>
                <CustomText style={[isCurrencyBold ? globalThemeStyle.text_Bold : globalThemeStyle.text_Regular, {fontSize: Math.ceil(fontSize / 1.5), color: globalColors.MAIN_DARK, marginLeft: Math.ceil(fontSize / 5)}]}>{currency.iso.toLocaleUpperCase()}</CustomText>
            </View>
        </View>
    )
}
