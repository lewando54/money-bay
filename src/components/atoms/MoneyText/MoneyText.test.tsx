import React from 'react'
import renderer from 'react-test-renderer'

import MoneyText from './MoneyText'
describe('MoneyText', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<MoneyText />)
        expect(tree).toMatchSnapshot()
    })
})
