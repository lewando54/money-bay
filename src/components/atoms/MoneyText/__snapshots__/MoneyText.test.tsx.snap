// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyText should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-end",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 16,
        },
        [
          {
            "color": "white",
            "fontFamily": "MulishRegular",
          },
          {
            "color": "#1B1D4D",
            "fontSize": 20,
          },
        ],
      ]
    }
  >
    10
    .
  </Text>
  <View
    style={
      [
        {
          "alignItems": "flex-end",
          "flexDirection": "row",
        },
        {
          "paddingBottom": 2,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishRegular",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 14,
            },
          ],
        ]
      }
    >
      00
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          [
            {
              "color": "white",
              "fontFamily": "MulishBold",
            },
            {
              "color": "#1B1D4D",
              "fontSize": 14,
              "marginLeft": 4,
            },
          ],
        ]
      }
    >
      USD
    </Text>
  </View>
</View>
`;
